public/js/app.js

```diff
@@ -1,22 +1,34 @@
-const weatherForm = document.querySelector('form')
 const search = document.querySelector('input')
 const weatherMessage = document.querySelector('#message')
 const weatherIcon = document.querySelector('#weatherIcon')
+const custLocWeather = document.querySelector('#custLoc')
+const currLocWeather = document.querySelector('#currLoc')
 
-weatherForm.addEventListener('submit', (event) => {
+custLocWeather.addEventListener('click', async (event) => {
     event.preventDefault()
     const location = search.value
     weatherMessage.textContent = 'Loading'
-    fetch(`/weather?address=${location}`).then((response) => {
-        response.json().then((data) => {
-            if (data.error) {
-                weatherMessage.textContent = data.error
-            } else {
-                weatherIcon.style.display = 'block'
-                weatherIcon.alt = `${data.description}`
-                weatherIcon.src = `${data.icon}`
-                weatherMessage.textContent = `You're in ${data.location}. The temperature is ${data.temperature}. And it feels like ${data.feelslike}.`
-            }
-        })
+    const response = await fetch(`/weather?address=${location}`)
+    const data = await response.json()
+    if (data.error) return weatherMessage.textContent = data.error
+        weatherIcon.style.display = 'block'
+        weatherIcon.alt = `${data.description}`
+        weatherIcon.src = `${data.icon}`
+        weatherMessage.textContent = `You're in ${data.location}. The temperature is ${data.temperature}. And it feels like ${data.feelslike}.`
+})
+
+currLocWeather.addEventListener('click', (event) => {
+    event.preventDefault()
+    if (!navigator.geolocation) {
+        return alert('Geolocation is not supported by your browser.')
+    }
+    navigator.geolocation.getCurrentPosition(async (position) => {
+        const response = await fetch(`/weather?lat=${position.coords.latitude}&lon=${position.coords.longitude}`)
+        const data = await response.json()
```

Comment:
here and above tone of code duplication. 
move data fetching to specific api client module and dry (don't repeat your-self) error displaying. We can discuss that on next call as well

Reply:
Fixed.

But probably this can be changed further, because I still got kind of the same code duplication issue as in:
https://github.com/sychova/nodeJS_course-weather_website/pull/1#discussion_r759592452